src/Mvc/Mvc.Core/test/ControllerBaseTest.cs

```diff
@@ -2290,6 +2291,171 @@ public void Controller_StatusCode_SetsStatusCode()
             Assert.Equal(statusCode, result.StatusCode);
         }
 
+        [Fact]
+        public void ValidationProblemDetails_UsesErrorDataConfiguredInApiBehaviorOptions()
+        {
+            // Arrange
+            var title = "Custom title";
+            var link = "http://custom-link";
+            var serviceProvider = new ServiceCollection()
+                .AddOptions()
+                .AddTransient<IConfigureOptions<ApiBehaviorOptions>, ApiBehaviorOptionsSetup>()
+                .Configure<ApiBehaviorOptions>(o => o.ClientErrorMapping[400] = new ClientErrorData { Link = link, Title = title })
+                .BuildServiceProvider();
+
+            var context = new ControllerContext(new ActionContext(
+                new DefaultHttpContext { RequestServices = serviceProvider },
+                new RouteData(),
+                new ControllerActionDescriptor()));
+
+            var controller = new TestableController
+            {
+                ControllerContext = context,
+            };
```

Comment:
See I don't feel like this amount of setup is reasonable for someone who wants to unit test a controller.